docs/build/cookbook.rst

```diff
@@ -1573,4 +1573,42 @@ the same ``env.py`` file can be invoked using asyncio as::
             await conn.run_sync(run_upgrade, config.Config("alembic.ini"))
 
 
-    asyncio.run(run_async_upgrade())
\ No newline at end of file
+    asyncio.run(run_async_upgrade())
+
+
+Data migrations
+===============
+
+Alembic migrations are designed for schema migrations.
+The nature of data migrations are inherently different and it's not in fact possible in the general case to write data migrations that integrate with Alembic's schema versioning model.
```

Comment:
let's be less definitive here:

> it's not in fact possible in the general case

it's not in fact advisable in the general case